File: Python/Program/KomponenEksternal.py
from Komponen import Komponen

class KomponenEksternal(Komponen):
    def __init__(self, merk="", nama="", harga=0, jenis_koneksi="", terhubung=False):
        super().__init__(merk, nama, harga)
        self.jenis_koneksi = jenis_koneksi
        self.terhubung = terhubung
    
    # Setter
    def set_jenis_koneksi(self, jenis_koneksi):
        self.jenis_koneksi = jenis_koneksi
    
    def set_terhubung(self, terhubung):
        self.terhubung = terhubung
    
    # Getter
    def get_jenis_koneksi(self):
        return self.jenis_koneksi
    
    def get_terhubung(self):
        return self.terhubung
    
    # Tampilkan Info
    def tampilkan_info(self):
        status = "Terhubung" if self.terhubung else "Tidak terhubung"
        return f"{super().tampilkan_info()} | Jenis Koneksi: {self.jenis_koneksi} | Status: {status}"

File: Java/Program/KomponenEksternal.java
public class KomponenEksternal extends Komponen {
    private String jenisKoneksi;
    private boolean terhubung;

    // Constructor
    public KomponenEksternal() {
        super();
        this.jenisKoneksi = "";
        this.terhubung = false;
    }

    public KomponenEksternal(String merk, String nama, int harga, String jenisKoneksi, boolean terhubung) {
        super(merk, nama, harga);
        this.jenisKoneksi = jenisKoneksi;
        this.terhubung = terhubung;
    }

    // Setter methods
    public void setJenisKoneksi(String jenisKoneksi) { this.jenisKoneksi = jenisKoneksi; }
    public void setTerhubung(boolean terhubung) { this.terhubung = terhubung; }

    // Getter methods
    public String getJenisKoneksi() { return jenisKoneksi; }
    public boolean getTerhubung() { return terhubung; }

    // Tampilkan Info
    @Override
    public String tampilkanInfo() {
        return super.tampilkanInfo() + " | Jenis Koneksi: " + jenisKoneksi + " | Status: " + (terhubung ? "Terhubung" : "Tidak terhubung");
    }
}